### PR TITLE
Shift Optimization

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

Reduce shift calculations to shifts that may have an actual result.

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

## Changelog

- [X] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: evm/src/main/java/org/hyperledger/besu/evm/operation/SarOperation.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public SarOperation(final GasCalculator gasCalculator) {
     } else {
       final int shiftAmountInt = shiftAmount.toInt();
 
-      if (shiftAmountInt >= 256) {
+      if (shiftAmountInt >= 256 || shiftAmountInt < 0) {
```
Comment (macfarla) [inline evm/src/main/java/org/hyperledger/besu/evm/operation/SarOperation.java:47]: what happens if it's == 0

Comment (macfarla) [reply evm/src/main/java/org/hyperledger/besu/evm/operation/SarOperation.java:47]: also would be great to have a test since this caused an issue

Comment (shemnon) [reply evm/src/main/java/org/hyperledger/besu/evm/operation/SarOperation.java:47]: The bytes handler handles that efficiently.

**Review by macfarla — APPROVED** [review]

File: evm/src/main/java/org/hyperledger/besu/evm/operation/SarOperation.java (hunk 1 above)
Comment (macfarla) [reply evm/src/main/java/org/hyperledger/besu/evm/operation/SarOperation.java:47]: 👍

**Pull request merged**